src/app/modules/workspace/__tests__/workspace.routes.spec.ts

```diff
@@ -75,14 +77,83 @@ describe('workspaces.routes', () => {
     })
 
     it('should return 500 when database errors occur', async () => {
+      const mockErrorMessage = 'something went wrong'
       jest
         .spyOn(WorkspaceModel, 'getWorkspaces')
-        .mockRejectedValueOnce(new Error('something went wrong'))
+        .mockRejectedValueOnce(new Error(mockErrorMessage))
       const response = await request.get(GET_WORKSPACES_ENDPOINT)
 
       expect(response.status).toEqual(500)
       expect(response.body).toEqual({
-        message: 'Something went wrong. Please try again.',
+        message: formatErrorRecoveryMessage(mockErrorMessage),
+      })
+    })
+  })
+
+  describe('POST /workspaces', () => {
+    const CREATE_WORKSPACE_ENDPOINT = '/workspaces'
+
+    it('should return 200 with created workspace on successful creation', async () => {
+      const createWorkspaceParam = {
+        title: 'validWorkspace',
+      }
+      const response = await request
+        .post('/workspaces')
+        .send(createWorkspaceParam)
+      const expected = {
+        title: createWorkspaceParam.title,
+        admin: MOCK_USER_ID.toHexString(),
+        formIds: [],
+      }
+
+      expect(response.status).toEqual(200)
+      expect(response.body).toMatchObject(expected)
+    })
+
+    it('should return 400 when workspace title is invalid', async () => {
+      const createInvalidWorkspaceParam = {
+        title: 'a',
+      }
+
+      const response = await request
+        .post(CREATE_WORKSPACE_ENDPOINT)
+        .send(createInvalidWorkspaceParam)
+
+      expect(response.status).toEqual(400)
+      expect(response.body).toEqual(
+        buildCelebrateError({
+          body: {
+            key: 'title',
+            message: '"title" length must be at least 4 characters long',
+          },
+        }),
+      )
+    })
+
+    it('should return 401 when user is not logged in', async () => {
+      await logoutSession(request)
+      const response = await request.post(CREATE_WORKSPACE_ENDPOINT)
+
+      expect(response.status).toEqual(401)
+      expect(response.body).toEqual({ message: 'User is unauthorized.' })
+    })
+
+    it('should return 500 when database errors occur', async () => {
```

Comment:
think we can probably separate this into the various database error cases, e.g. 409 for conflict error, 422 for validation error

Reply:
I've split it to the various error cases, and tested in the controller